code/modules/vending/_vending.dm

```diff
@@ -646,11 +660,18 @@ GLOBAL_LIST_EMPTY(vending_products)
 	. = TRUE
 	if(!user.transferItemToLoc(I, src))
 		return FALSE
+	to_chat(user, span_notice("You insert [I] into [src]'s input compartment."))
+
+	for(var/datum/data/vending_product/product_datum in product_records + coin_records + hidden_records)
+		if(ispath(I, product_datum.product_path))
```

Comment:
Think this needs to be I.type because it doesn't work in testing

Reply:
Also, although it doesn't hurt to have a second check, istype() was fine here because the item had to pass canLoadItem() which checks the item's type was in the original product lists.